features/nfc/nfc/NFCEEPROMDriver.h

```diff
@@ -0,0 +1,188 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_NFC_EEPROM_DRIVER_H
+#define MBED_NFC_EEPROM_DRIVER_H
+
+#include <stdint.h>
+
+#include "NFCDefinitions.h"
+#include "NFCTarget.h"
+#include "NFCEEPROMDriver.h"
+
+namespace mbed {
+namespace nfc {
+
+    /**
+     * @addtogroup nfc
+     * @{
+     */
+    
+    /**
+     * The abstraction for a NFC EEPROM driver.
+     * Implementers need to derive from this class and implement its methods.
+     */ 
+    class NFCEEPROMDriver  {
+    public:
+        /**
+         * Construct a NFCEEPROM driver instance.
+         */
+        NFCEEPROMDriver();
+        
+        /**
+         * The NFCEEPROMDriver delegate.
+         * Methods in this class are called by the driver on completion of long-running operations.
+         */
+        struct Delegate {
+            /**
+             * Completion of session start operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             */  
+            virtual void has_started_session(bool success) = 0;
```

Comment:
all delegates need an empty implementation so that the user can overwritten only the ones he's interested in

Reply:
Actually this the PAL layer, empty implementations should be provided on user side, these can remain abstract